src/asset-swapper/utils/protocol_fee_utils.ts

```diff
@@ -42,15 +42,16 @@ export class ProtocolFeeUtils {
     }
 
     /** @returns gas price (in wei) */
-    public async getGasPriceEstimationOrThrowAsync(shouldHardRefresh?: boolean): Promise<BigNumber> {
-        if (this._gasPriceEstimation.eq(constants.ZERO_AMOUNT)) {
-            return this._getGasPriceFromGasStationOrThrowAsync();
-        }
-        if (shouldHardRefresh) {
-            return this._getGasPriceFromGasStationOrThrowAsync();
-        } else {
-            return this._gasPriceEstimation;
+    public async getGasPriceEstimationOrThrowAsync(defaultGasPrices: GasPrices): Promise<GasPrices> {
```

Comment:
As it is it can fail even when `defaultGasPrices` is provided which seems weird. 

How about providing two different methods:
`getGasPriceEstimationOrThrowAsync()` -- no default
`getGasPriceEstimationOrDefault()` -- fallback to provided default.